File: codility/src/main/java/codility/com/lessons/l02arrays/OddOccurrencesInArray.java
package codility.com.lessons.l02arrays;

import java.util.HashSet;
import java.util.Set;

/**
 * https://app.codility.com/programmers/lessons/2-arrays/odd_occurrences_in_array/
 * 100% 
 * @author 08115341622
 */
public class OddOccurrencesInArray {
	public int solution(int[] A) {
		Set<Integer> elementos = new HashSet<>();
		for (int x : A) {
			if (!elementos.remove(x)) {
				elementos.add(x);
			}
		}
		return elementos.iterator().next();
    }
}

File: codility_new/odd_occurrences_in_array.py
# https://app.codility.com/programmers/lessons/2-arrays/odd_occurrences_in_array/
def solution(A):
    count_numbers = {}
    for x in A:
        count_numbers[x] = count_numbers.get(x, 0) + 1
    
    for number, count in count_numbers.items():
        if count % 2 == 1:
            return number
    
    return 0
